src/main/java/org/folio/search/repository/classification/InstanceClassificationJdbcRepository.java

```diff
@@ -0,0 +1,164 @@
+package org.folio.search.repository.classification;
+
+import static org.folio.search.utils.JdbcUtils.getQuestionMarkPlaceholder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.PreparedStatement;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.search.model.index.InstanceSubResource;
+import org.folio.search.utils.JdbcUtils;
+import org.folio.spring.FolioExecutionContext;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Log4j2
+@Repository
+@RequiredArgsConstructor
+public class InstanceClassificationJdbcRepository implements InstanceClassificationRepository {
+
+  private static final String INSTANCE_CLASSIFICATION_TABLE_NAME = "instance_classification";
+  private static final String CLASSIFICATION_TYPE_COLUMN = "classification_type";
+  private static final String CLASSIFICATION_NUMBER_COLUMN = "classification_number";
+  private static final String TENANT_ID_COLUMN = "tenant_id";
+  private static final String INSTANCE_ID_COLUMN = "instance_id";
+  private static final String SHARED_COLUMN = "shared";
+  private static final String CLASSIFICATION_TYPE_DEFAULT = "<null>";
+
+  private static final String SELECT_ALL_SQL = "SELECT * FROM %s;";
+  private static final String SELECT_ALL_BY_INSTANCE_ID_AGG = """
+    SELECT
+        t1.classification_number,
+        t1.classification_type,
+        json_agg(json_build_object(
+            'instanceId', t1.instance_id,
+            'shared', t1.shared,
+            'tenantId', t1.tenant_id
+        )) AS instances
+    FROM %1$s t1
+    INNER JOIN %1$s t2 ON t1.classification_number = t2.classification_number
+                                        AND t1.classification_type = t2.classification_type
+                                        AND t2.instance_id IN (%2$s)
+    GROUP BY t1.classification_number, t1.classification_type;
+    """;
+  private static final String INSERT_SQL = """
+    INSERT INTO %s (classification_type, classification_number, tenant_id, instance_id, shared)
+    VALUES (?, ?, ?, ?, ?)
+    ON CONFLICT (classification_type, classification_number, tenant_id, instance_id)
+    DO UPDATE SET shared = EXCLUDED.shared;
```

Comment:
Shouldn't we also update type as it is nullable and I guess could be updated?

Reply:
If type could be updated - then I guess we would need to left a note to delete old document before inserting the new one as id changes, in elasticsearch implementation story

Reply:
classification_type is part ok PK for identifing uniqueness of classification. From perspective of UI classification_number is also updatable. In case of some of these fields updated - it's identified as new classification and old one with previous values will be deleted.